### PR TITLE
update to go 1.23

### PR DESCRIPTION
**Pull request by crazy-max**
similar to:
* https://github.com/moby/buildkit/pull/5492
* https://github.com/moby/buildkit/pull/5314
* https://github.com/moby/buildkit/pull/5484

### CODE REVIEW

**Comment by thaJeztah** [comment]
Can you update the order of the commits? 

- The `fix golangci-lint issues` should probably be first
- then the "update golangci", and then the other ones.
- oh, and the `G115  # integer overflow conversion (TODO: verify these)` should be in the second one (as it's related to the golangci-lint update)

**Comment by thompson-shaun** [comment]
I _think_ everything requested was addressed @thaJeztah - are we OK to merge this one?

File: hack/dockerfiles/lint.Dockerfile [hunk 1]
```diff
@@ -1,12 +1,11 @@
 # syntax=docker/dockerfile:1
 
-ARG GO_VERSION=1.22
-ARG XX_VERSION=1.3.0
-ARG GOLANGCI_LINT_VERSION=1.57.2
-ARG GOPLS_VERSION=v0.20.0
+ARG GO_VERSION=1.23
+ARG XX_VERSION=1.5.0
+ARG GOLANGCI_LINT_VERSION=1.61.0
```
Comment (tonistiigi) [inline hack/dockerfiles/lint.Dockerfile:5]: Why 61 in 62 is the latest?

Comment (crazy-max) [reply hack/dockerfiles/lint.Dockerfile:5]: Was released after my PR got opened :sweat_smile:

**Comment by thaJeztah** [comment]
Looks like the first commit doesn't work; I'm guessing the older golangci-lint is not compatible with go1.23, so to be "correct", the go bump should probably go after the golangci-lint bump

**Comment by crazy-max** [comment]
> Looks like the first commit doesn't work; I'm guessing the older golangci-lint is not compatible with go1.23, so to be "correct", the go bump should probably go after the golangci-lint bump

Done

File: hack/dockerfiles/lint.Dockerfile (hunk 1 above)
Comment (thaJeztah) [reply hack/dockerfiles/lint.Dockerfile:5]: As you're updating; might as well update this one as well (if it helps; I didn't need to make changes in moby when updating)

Comment (crazy-max) [reply hack/dockerfiles/lint.Dockerfile]: Done

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by crazy-max** [comment]
Ah fixes needed with latest golangci-lint 1.62 actually: https://github.com/docker/buildx/actions/runs/11930900417/job/33252637019?pr=2782#step:5:368

```
#19 [lint 1/1] RUN --mount=target=/go/src/github.com/docker/buildx     --mount=target=/root/.cache,type=cache,id=lint-cache-darwin/arm64     xx-go --wrap &&     golangci-lint run
#19 92.91 driver/remote/driver.go:51:8: lostcancel: the cancel function returned by context.WithTimeoutCause should be called, not discarded, to avoid a context leak (govet)
#19 92.91 		ctx, _ := context.WithTimeoutCause(cancelCtx, 20*time.Second, errors.WithStack(context.DeadlineExceeded))
#19 92.91 		     ^
#19 92.91 builder/builder.go:526:14: lostcancel: the cancel function returned by context.WithTimeoutCause should be called, not discarded, to avoid a context leak (govet)
#19 92.91 	timeoutCtx, _ := context.WithTimeoutCause(cancelCtx, 20*time.Second, errors.WithStack(context.DeadlineExceeded))
#19 92.91 	            ^
#19 92.91 commands/inspect.go:39:14: lostcancel: the cancel function returned by context.WithTimeoutCause should be called, not discarded, to avoid a context leak (govet)
#19 92.91 	timeoutCtx, _ = context.WithTimeoutCause(timeoutCtx, 20*time.Second, errors.WithStack(context.DeadlineExceeded))
#19 92.91 	            ^
#19 92.91 commands/ls.go:62:14: lostcancel: the cancel function returned by context.WithTimeoutCause should be called, not discarded, to avoid a context leak (govet)
#19 92.91 	timeoutCtx, _ = context.WithTimeoutCause(timeoutCtx, 20*time.Second, errors.WithStack(context.DeadlineExceeded))
#19 92.91 	            ^
#19 92.91 commands/rm.go:154:14: lostcancel: the cancel function returned by context.WithTimeoutCause should be called, not discarded, to avoid a context leak (govet)
#19 92.91 	timeoutCtx, _ = context.WithTimeoutCause(timeoutCtx, 20*time.Second, errors.WithStack(context.DeadlineExceeded))
#19 92.91 	            ^
```

**Pull request merged**